### ISSUE TITLE
AIShell results for Conformer not reproducible

### ISSUE THREAD

**Issue by jinpoon** [espnet/espnet#3611]
Hi team, 
I am following the instructions on the aishell [README.md](https://github.com/espnet/espnet/blob/master/egs2/aishell/asr1/README.md) and trying to reproduce the conformer results.  I downloaded the data from the link (https://zenodo.org/record/4105763#.X40xe2j7QUE). The instructions ask to checkout the commit `20b0c89369d9dd3e05780b65fdd00a9b4f4891e5`, but in this commit `./conf/tuning/train_asr_conformer.yaml` doesn't exist. I am wondering whether that's the correct commit.
With the provided model, on [master branch](https://github.com/espnet/espnet/tree/0f7558a716ab830d0c29da8785840124f358d47b), I got 8.0%/9.4% for dev/test. Is the decode config up to date? Could you share the decode configuration as well? Thank you.

My environment:
python: 3.8.11
pytorch version: 1.8.0

**Comment by sw005320** [comment]
@pengchengguo, can you answer it for me?
I'm expecting the compatibility issue of the relative positional encoding.

**Comment by pengchengguo** [comment]
Hi @jinpoon,
Thanks for your report. Maybe it is caused by the compatibility issue of the relative positional encoding as @sw005320 mentioned. I will check it ASAP. 

Before fixing it, you can try different relative position encodings by modifying the `rel_pos_type` in the `config.yaml` file like `rel_pos_type: legacy` or `rel_pos_type: latest`.

**Comment by jinpoon** [comment]
@pengchengguo thanks for the reply. I am on a [commit](https://github.com/espnet/espnet/tree/70d16d210cdce28e71b7892a0ec96eaaa6474d64) that doesn't support `rel_pos_type` option, while the decode for the librispeech conformer model could match the reported values. Does it mean that positional encoding is probably not the main cause?

**Comment by jasonyong** [comment]
Meet the same problem!

I download the same res in https://zenodo.org/record/4105763#.X40xe2j7QUE, use the config asr_train_asr_conformer3_raw_char_batch_bins4000000_accum_grad4_sp/config.yaml to decode.

Decoding param is beam:10, ctc_weight=0.4, the result is very wrong:
HYP: 房产市场将有过度上涨的
REF: 甚至出现交易几乎停滞的情况

When I print the feature extracted by frontend module, that is:
(Pdb) feats
tensor([[[14.6090, 10.5855, 10.6156,  ...,  4.5462,  4.0073,  4.3834],
         [14.5980, 13.0177, 10.9505,  ...,  3.8481,  3.7526,  3.2396],
         [13.9000, 12.7130,  9.9016,  ...,  3.0969,  3.5763,  1.8216],
         ...,
         [15.8055, 11.3160, 10.1481,  ...,  2.1864,  1.8911,  1.4883],
         [15.5125, 10.3162, 11.0743,  ...,  2.2568,  2.1623,  1.5548],
         [14.8051, 13.5617, 12.8094,  ...,  2.7859,  3.3965,  2.1815]]],
       device='cuda:0')
But the global feature mean calc by asr_stats_raw_sp/train/feats_stats.npz is:
(Pdb) self.normalize.mean
tensor([ -6.3421,  -7.3554,  -7.3735,  -6.9312,  -6.1869,  -6.2125,  -6.7251,
         -7.2656,  -7.7793,  -7.8812,  -8.0074,  -8.2777,  -8.5129,  -8.8453,
         -9.1373,  -9.4340,  -9.8731, -10.0165, -10.1411, -10.2466, -10.2263,
        -10.5242, -10.6074, -10.7111, -10.8452, -10.7647, -10.9943, -11.1065,
        -11.0815, -11.0649, -11.3850, -11.1433, -11.4263, -11.3008, -11.3805,
        -11.3902, -11.4292, -11.5012, -11.5717, -11.6447, -11.7449, -11.8608,
        -11.9931, -12.0650, -12.1762, -12.2386, -12.3090, -12.3092, -12.3272,
        -12.3353, -12.3411, -12.3453, -12.3919, -12.4122, -12.4890, -12.5530,
        -12.6825, -12.7859, -12.9135, -13.1278, -13.3277, -13.5078, -13.6897,
        -13.8625, -13.9801, -14.0923, -14.1703, -14.1709, -14.1837, -14.1838,
        -14.1635, -14.1466, -14.1328, -14.0918, -14.0970, -14.1503, -14.5603,
        -15.9254, -16.5352, -17.3522], device='cuda:0')

That is very different range between feature and mean, one is postive and another one is negtive.
Maybe the res file is wrong, @sw005320 @pengchengguo  please the check file: asr_stats_raw_sp/train/feats_stats.npz, thanks!

**Comment by jasonyong** [comment]
Hi @pengchengguo ,

Thanks for you replay. I have change the code to this commit (https://github.com/espnet/espnet/commit/161b1f2d22b84dc42fc5ce694de9e1d260e65e5d), but the result is the same.

Can @kamo-naoyuki figure that out?

By the way, @pengchengguo what's your best result on aishell-1 without LM, currently my best result is 5.2 on test, use the same structure of  the default config, with utt_cmvn, 80 dim logmel fbank as input.

**Comment by pengchengguo** [comment]
Hi @jasonyong, I will decode w/o LM and post the results later.

**Comment by pengchengguo** [comment]
Hi @jasonyong, here are my results:

### CER

|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
|---|---|---|---|---|---|---|---|---|
|decode_asr_rnn_asr_model_valid.acc.ave/dev|14326|205341|95.7|4.2|0.1|0.1|4.4|34.4|
|decode_asr_rnn_asr_model_valid.acc.ave/test|7176|104765|95.3|4.6|0.1|0.1|4.8|36.7|
|decode_asr_rnn_lm_lm_train_lm_transformer_zh_char_valid.loss.ave_asr_model_valid.acc.ave/dev|14326|205341|95.5|4.4|0.1|0.1|4.6|34.9|
|decode_asr_rnn_lm_lm_train_lm_transformer_zh_char_valid.loss.ave_asr_model_valid.acc.ave/test|7176|104765|95.2|4.7|0.1|0.1|4.9|37.0|

I use the default conformer config files except for some changes related to batch size (e.g. batch_bins=12000000, accum_grad=1) since the model is trained on 4 A100 GPUs.
It is strange that w/o LM shows better performance. Maybe I should tune some LM decoding parameters.

**Comment by jasonyong** [comment]
Hi @pengchengguo ,

Thanks for posting your results. 

So when using the default config, your feature input is raw(this is depending on data-loader), right?

**Comment by pengchengguo** [comment]
Yes, I use the raw waves as inputs and the fbank features will be extracted by `espnet/espnet2/asr/frontend/default.py` component.

**Comment by jasonyong** [comment]
Got it. Thank you!
